src/api.cc

```diff
@@ -119,31 +121,11 @@ AkliteClient::~AkliteClient() {
 
 static bool compareTargets(const TufTarget& a, const TufTarget& b) { return a.Version() < b.Version(); }
 
-CheckInResult AkliteClient::CheckIn() const {
-  if (!configUploaded_) {
-    client_->reportAktualizrConfiguration();
-    configUploaded_ = true;
-  }
-  client_->reportNetworkInfo();
-  client_->reportHwInfo();
-  client_->reportAppsState();
-
-  auto status = CheckInResult::Status::Ok;
-  Uptane::HardwareIdentifier hwidToFind(client_->config.provision.primary_ecu_hardware_id);
-
-  LOG_INFO << "Refreshing Targets metadata";
-  const auto rc = client_->updateImageMeta();
-  if (!std::get<0>(rc)) {
-    LOG_WARNING << "Unable to update latest metadata, using local copy: " << std::get<1>(rc);
-    if (!client_->checkImageMetaOffline()) {
-      LOG_ERROR << "Unable to use local copy of TUF data";
-      return CheckInResult(CheckInResult::Status::Failed, "", {});
-    }
-    status = CheckInResult::Status::OkCached;
-  }
-
+// Returns a sorted list of targets matching tags and hwid (or one of secondary_hwids)
+std::vector<TufTarget> AkliteClient::filterTargets(const std::vector<Uptane::Target> allTargets,
```

Comment:
It looks like non-class static and internal function.

Reply:
I've converted it now to a non-class static function. Force-push coming soon.

Reply:
Done